examples/query/synth_data_llm_filter.py

```diff
@@ -0,0 +1,169 @@
+            - 'properties': A dictionary of properties associated with the element.
+            - 'text_representation': The text representation of the element.
+    """
+#    return Element(text_representation=text)
+    return {'properties': {}, 'text_representation': text}
+
+def generate_elements(keywords, probabilities, max_elements):
+
+    strings = []
+    num = random.randint(1, max_elements)
+    base_text = """
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed auctor in enim sit amet tristique. 
+    In eu egestas ex. Maecenas porttitor gravida libero ac porttitor. Morbi elementum eleifend nisl.
+     Nullam auctor magna nec mollis ultricies. Mauris non pulvinar turpis. Pellentesque dictum 
+     tortor ac elit feugiat varius. Nulla sollicitudin, lectus non venenatis finibus, nulla ex 
+     tempor urna, id ornare est nisl et lacus. Pellentesque commodo turpis in nibh aliquet viverra. 
+     Curabitur varius, elit nec finibus scelerisque, nibh eros congue dui, et ultrices justo magna 
+     eu sem. Sed ultricies posuere posuere. Ut euismod neque at diam interdum, sed rhoncus quam 
+     iaculis. Morbi sit amet tempus orci, a dictum metus. Donec a imperdiet sem. Vestibulum tincidunt 
+     gravida turpis vitae aliquet.
+    """
+    for i in range(num):
+        text = "base_text" + str(i)
+        selected_keywords = ""
+        for keyword, probability in zip(keywords, probabilities):
+            if random.random() < probability:
+                selected_keywords = selected_keywords + " " + keyword
+        strings.append(make_element(text+selected_keywords))
+    return strings
+
+def generate_doc(doc_id, keywords, probabilities, max_elems):
+    """
+    Generate a document with random elements.
+
+    Args:
+        keywords (list): List of keywords.
+        probabilities (list): List of probabilities corresponding to the keywords.
+        max_elems (int): Maximum number of elements in each document.
+
+    Returns:
+        A Document objects with a random set of elements.
+    """
+    return Document (doc_id=doc_id, elements=generate_elements(keywords, probabilities, max_elems))
+#    return {'doc_id':doc_id, 'elements':generate_elements(keywords, probabilities, max_elems)}
+
+
+def generate_docset(keywords, probabilities, max_docs, max_elems):
+    """
+    Generate a document set with random elements.
+
+    Args:
+        keywords (list): List of keywords.
+        probabilities (list): List of probabilities corresponding to the keywords.
+        max_docs (int): Maximum number of documents in the docset.
+        max_elems (int): Maximum number of elements in each document.
+
+    Returns:
+        list: A list of Document objects, each containing a random set of elements.
+    """
+    docset = []
+    for i in range(max_docs):
+        doc = generate_doc(str(i), keywords, probabilities, max_elems)
+        docset.append(doc)
+    return docset
+
+
+keywords = ['cat', 'dog']
```

Comment:
Weird to have this as two parallel arrays. It makes it hard to be sure that the counts are the same and which keyword goes with which name. 
Make this a dictionary.
```
keyword_to_probability = {
   'cat": 0.1,
   'dog': 0.2,
}
```